src/user-lists/user-list-stat.service.ts

```diff
@@ -0,0 +1,442 @@
+import { Injectable, NotFoundException } from "@nestjs/common";
+import { Repository, SelectQueryBuilder } from "typeorm";
+import { InjectRepository } from "@nestjs/typeorm";
+
+import { PageDto } from "../common/dtos/page.dto";
+import { PageMetaDto } from "../common/dtos/page-meta.dto";
+import { DbUserListContributor } from "./entities/user-list-contributor.entity";
+import { DbUserListContributorStat } from "./entities/user-list-contributor-stats.entity";
+import {
+  UserListContributorStatsOrderEnum,
+  UserListContributorStatsTypeEnum,
+  UserListMostActiveContributorsDto,
+} from "./dtos/most-active-contributors.dto";
+import { ContributionsTimeframeDto } from "./dtos/contributions-timeframe.dto";
+import { DbContributionStatTimeframe } from "./entities/contributions-timeframe.entity";
+import { DbContributionsProjects } from "./entities/contributions-projects.entity";
+import { DbContributorCategoryTimeframe } from "./entities/contributors-timeframe.entity";
+
+@Injectable()
+export class UserListStatsService {
+  constructor(
+    @InjectRepository(DbUserListContributor, "ApiConnection")
+    private userListContributorRepository: Repository<DbUserListContributor>
+  ) {}
+
+  baseQueryBuilder(): SelectQueryBuilder<DbUserListContributor> {
+    const builder = this.userListContributorRepository.createQueryBuilder("user_list_contributors");
+
+    return builder;
+  }
+
+  async findListContributorStatsByProject(listId: string, repoId: number): Promise<DbUserListContributorStat[]> {
+    const queryBuilder = this.baseQueryBuilder();
+
+    queryBuilder.innerJoin("users", "users", "user_list_contributors.user_id=users.id");
+
+    queryBuilder
+      .select("users.login", "login")
+      .andWhere("user_list_contributors.list_id = :listId", { listId })
+      .addSelect(
+        `(
+          SELECT SUM("pull_requests"."commits")
+          FROM "pull_requests"
+          WHERE "pull_requests"."author_login" = "users"."login"
+            AND "pull_requests"."repo_id" = ${repoId}
+        )::INTEGER`,
+        "commits"
+      )
+      .addSelect(
+        `(
+          SELECT COALESCE(COUNT("pull_requests"."id"), 0)
+          FROM "pull_requests"
+          WHERE "pull_requests"."author_login" = "users"."login"
+            AND "pull_requests"."repo_id" = ${repoId}
+        )::INTEGER`,
+        "prsCreated"
```

Comment:
Should match the snake case for the entity

```suggestion
        "prs_created"
```